### ISSUE TITLE
Kubuntu 22.04 - teams-for-linux 1.52  snap generates many apparmor DENIED operations 

### ISSUE THREAD

**Issue by rbeldin** [IsmaelMartinez/teams-for-linux#1281]
**Describe the bug**
When starting snap teams-for-linux on Kubuntu 22.04 (fully patched)  I see: 

systemd[5210]: Started snap.teams-for-linux.teams-for-linux-07adcd7e-797e-4938-96bf-62690e304b66.scope.
kernel: [48265.914430] kauditd_printk_skb: 1 callbacks suppressed
kernel: [48265.914436] audit: type=1400 audit(1717613996.233:83260): apparmor="DENIED" operation="capable" profile="/usr/lib/snapd/snap-confine" pid=48572 comm="snap-confine" capability=12  capname="net_admin"
kernel: [48265.914447] audit: type=1400 audit(1717613996.233:83261): apparmor="DENIED" operation="capable" profile="/usr/lib/snapd/snap-confine" pid=48572 comm="snap-confine" capability=38  capname="perfmon"
systemd[5209]: Started Teams for Linux.
kernel: [48265.924153] audit: type=1400 audit(1717613996.241:83262): apparmor="DENIED" operation="open" profile="snap.teams-for-linux.teams-for-linux" name="/home/username/.bashrc" pid=48572 comm="command.sh" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
kernel: [48265.924969] audit: type=1400 audit(1717613996.245:83263): apparmor="DENIED" operation="open" profile="snap.teams-for-linux.teams-for-linux" name="/home/username/.bashrc" pid=48572 comm="desktop-init.sh" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
kernel: [48265.926870] audit: type=1400 audit(1717613996.245:83264): apparmor="DENIED" operation="open" profile="snap.teams-for-linux.teams-for-linux" name="/home/username/.bashrc" pid=48572 comm="desktop-common." requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
kernel: [48265.943984] audit: type=1400 audit(1717613996.261:83265): apparmor="DENIED" operation="open" profile="snap.teams-for-linux.teams-for-linux" name="/home/username/.bashrc" pid=48572 comm="desktop-gnome-s" requested_mask="r" denied_mask="r" fsuid=1000 ouid=1000
kernel: [48265.970249] audit: type=1326 audit(1717613996.289:83267): auid=1000 uid=1000 gid=1000 ses=3 subj=snap.teams-for-linux.teams-for-linux pid=48647 comm="teams-for-linux" exe="/snap/teams-for-linux/602/teams-for-linux" sig=0 arch=c000003e syscall=330 compat=0 ip=0x7f7c5cafdf6b code=0x50000
kernel: [48265.970255] audit: type=1326 audit(1717613996.289:83266): auid=1000 uid=1000 gid=1000 ses=3 subj=snap.teams-for-linux.teams-for-linux pid=48646 comm="teams-for-linux" exe="/snap/teams-for-linux/602/teams-for-linux" sig=0 arch=c000003e syscall=330 compat=0 ip=0x7f17406fdf6b code=0x50000
kernel: [48265.994074] audit: type=1326 audit(1717613996.313:83268): auid=1000 uid=1000 gid=1000 ses=3 subj=snap.teams-for-linux.teams-for-linux pid=48572 comm="teams-for-linux" exe="/snap/teams-for-linux/602/teams-for-linux" sig=0 arch=c000003e syscall=425 compat=0 ip=0x7f84f08f588d code=0x50000
kernel: [48265.994086] audit: type=1326 audit(1717613996.313:83269): auid=1000 uid=1000 gid=1000 ses=3 subj=snap.teams-for-linux.teams-for-linux pid=48572 comm="teams-for-linux" exe="/snap/teams-for-linux/602/teams-for-linux" sig=0 arch=c000003e syscall=425 compat=0 ip=0x7f84f08f588d code=0x50000
dbus-daemon[5223]: apparmor="DENIED" operation="dbus_method_call"  bus="session" path="/modules/kwalletd5" interface="org.kde.KWallet" member="isEnabled" mask="send" name="org.kde.kwalletd5" pid=48572 label="snap.teams-for-linux.teams-for-linux" peer_pid=5233 peer_label="unconfined"
dbus-daemon[5223]: apparmor="DENIED" operation="dbus_method_call"  bus="session" path="/modules/kwalletd5" interface="org.kde.KWallet" member="close" mask="send" name="org.kde.kwalletd5" pid=48572 label="snap.teams-for-linux.teams-for-linux" peer_pid=5233 peer_label="unconfined"

uname -r
5.15.0-107-generic

**To Reproduce**

Login to kde desktop
Start teams-for-linux
Watch /var/log/syslog 

**Expected behavior**
Not get apparmor violations or describe method to suppress - with risks. 

**Screenshots**

**Desktop (please complete the following information):**
 - Kubuntu 22.04
 - Installation package snap teams-for-linux
 - Version 1.52

# snap info teams-for-linux
name:      teams-for-linux
summary:   Teams for Linux
publisher: Ismael Martinez Ramos (ismaelmartinez)
store-url: https://snapcraft.io/teams-for-linux
contact:   https://github.com/IsmaelMartinez/teams-for-linux
license:   unset
description: |
  Unofficial Microsoft Teams client for Linux using Electron. It uses the Web App and wraps it as a
  standalone application using Electron.
  
  If you encounter any issues, please feel free to reach out to us through the GitHub repository
  provided in the contact information. We will do our best to try and help you.
commands:
  - teams-for-linux
snap-id:      IIAJhVGfBhm393UTagoW5awtZlPDGqHd
tracking:     latest/stable
refresh-date: yesterday at 12:51 PDT
channels:
  latest/stable:    1.5.2 2024-06-04 (602) 89MB -
  latest/candidate: ↑                           
  latest/beta:      ↑                           
  latest/edge:      1.5.3 2024-06-04 (604) 90MB -
installed:          1.5.2            (602) 89MB -

**Debug**
When possible, please run the application from the terminal using `--webDebug` and try to reproduce the error.

The provide in this section the output from both the terminal and the browser debug console.

```bash
teams-for-linux --webDebug --appLogLevels=error,info,warn,debug
```

**Additional context**

**Comment by IsmaelMartinez** [comment]
hi @rbeldin , Snap installations tend to be a pain in some cases. Can you use another packaging system like `deb` ? 

Otherwise my suspicion is on the snap/apparmor configuration . I would check in the snap forums. 

I had a similar problem in the past and it was because my user home was not in the /home/user.name but in another path.

**Comment by rbeldin** [comment]
The deb appears to be gone.   I had it at one point in the past, but then was pointed to the snap.   There's also a flatpak for Fedora-like systems.   

The problem clearly seems to be with the snap policies and seems specific to this snap.  Other snaps don't give me this issue.  

This is as close as I can find to an official reference to the deb removal:

https://askubuntu.com/questions/1457083/cant-download-microsoft-teams-deb-file

**Comment by IsmaelMartinez** [comment]
That is the official deb app. MS removed official support for linux a while back. 

You can find the deb package in the releases section, or there are instructions in https://github.com/IsmaelMartinez/teams-for-linux?tab=readme-ov-file#debianubuntu-and-other-derivatives into how to configure it in your system.

**Comment by rbeldin** [comment]
Ah thanks for the clarification.  I got vectored to the snap some time ago and have been using that pretty happily but there are some quirks to using the snaps.    I just installed teams-for-linux_1.5.3_amd64.deb and that seems happy.  I can be as productive with that as teams lets me be.  :) 

Aside - why bother with snaps and flatpaks when the appImage works just fine?   They were all the rage a few years ago but the interest in them seems to have waned.
